src/views/product-tutorials-view/helpers/detect-and-reformat-learn-url.ts

```diff
@@ -0,0 +1,35 @@
+import { isThemeOrProduct } from 'lib/learn-client/types'
+import isAbsoluteUrl from 'lib/is-absolute-url'
+import { rewriteTutorialsLink } from 'lib/remark-plugins/rewrite-tutorial-links'
+import { getTutorialMap } from 'lib/remark-plugins/rewrite-tutorial-links/utils'
+
+let TUTORIAL_MAP
+
+/**
+ * Given a URL string,  detect whether it is a collection or tutorial URL
+ * in the format used on learn.hashicorp.com.
+ *
+ * If it is, then reformat the URL to work with dev-dot's URL structure.
+ * Otherwise, return the URL unmodified.
+ */
+async function detectAndReformatLearnUrl(url: string): Promise<string> {
+  // Don't mess with absolute URLs
+  if (isAbsoluteUrl(url)) {
+    return url
+  }
+  // Determine from path parts whether this is a collection or tutorial URL
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_beforeSlash, part1, part2] = url.split('/')
+  const isTutorialUrl = part1 === 'tutorials' && isThemeOrProduct(part2)
```

Comment:
There's another case here which is the product hub page link (i.e. /waypoint --> /waypoint/tutorials). We don't need to add it here but maybe we could add a few examples in the comments at the top of the file for exactly which urls will be rewritten here as a paper trail for when we use this function again later on.

Reply:
For context, `rewriteTutorialLinks` _will_ handle the product hub page link if its passed.

Reply:
ah yeah great call... i think i can quickly patch this up, I think it'd be like `isHubPageUrl = isThemeOrProduct(part1)`, and then we'd be able to pass that right to `rewriteTutorialLinks` 🙌 will give that a shot, and add a test case!

Reply:
✅ awesome, i think the above worked - now handles hub urls as well! going to merge this in 👍